fuzzywuzzy/fuzz.py

```diff
@@ -40,33 +40,22 @@
 # Basic Scoring Functions #
 ###########################
 
+@utils.check_for_none
+@utils.check_empty_string
 def ratio(s1, s2):
-
-    if s1 is None:
-        raise TypeError("s1 is None")
-    if s2 is None:
-        raise TypeError("s2 is None")
     s1, s2 = utils.make_type_consistent(s1, s2)
 
-    if len(s1) == 0 or len(s2) == 0:
```

Comment:
Yes, I did notice that, I forgot to talk to you about it.

I know the length can change when you do a type conversion, but I'm not aware of any situation where after conversion the length becomes 0 unless the original length was already 0. AFAIK this should never happen but I know string encoding sometimes has this kind of gotchas, so if you are aware of any situation where this could be a concern please let me know